modules/swagger-parser/src/test/java/io/swagger/parser/FileReferenceTests.java

```diff
@@ -11,12 +11,16 @@
 public class FileReferenceTests {
     @Test
     public void testIssue306() {
-        SwaggerDeserializationResult result = new SwaggerParser().readWithInfo("./src/test/resources/nested-file-references/issue-306.yaml", null, true);
+        SwaggerDeserializationResult result = new SwaggerParser().readWithInfo(getClass().getResource("/nested-file-references/issue-306.yaml").getFile(), null, true);
         assertNotNull(result.getSwagger());
 
         Swagger swagger = result.getSwagger();
 
-        assertTrue(swagger.getDefinitions().size() == 3);
+        assertEquals(swagger.getDefinitions().size(), 5);
```

Comment:
To be reviewed with attention.

The only things that has changed is here. I didn't understand why this test was passing before. As far as I understand, it's normal to find the `Animal` and `Category` definitions in the swagger, but I could be wrong.
I updated the assertions in this way.